### ISSUE TITLE
Issues of Extract operation in preview

### ISSUE THREAD

**Issue by slhh** [openstreetmap/iD#6674]
There are some issues with the extract operation in the current preview:

- the POI to be placed does not always follow the mouse pointer directly, but often with a random offset.
- the POI to be placed is invisible while the mouse is over the toolbar. The user might assume the operation is already complete.
- when pressing ESC before placing the POI this results in 2 changes to be saved but no undo available.
- the favorites should not be shown before the operation is complete.
- the safe button should better be replaced with cancel (like the drawing mode) until the POI is placed.

**Comment by quincylvania** [comment]
I've noticed some of these issues and I think my preferred solution would be to instantly place the point in the middle of the feature (or as close as possible in-frame). The user can then move the point as needed. This seems simpler than building out a workflow, plus it's easier on touchscreens.

**Comment by quincylvania** [comment]
> my preferred solution would be to instantly place the point in the middle of the feature

I made it work like this. It's definitely simpler, consistent with the related Combine and Downgrade operations, and touchscreen friendly. Plus it automatically puts points in the center of areas, so mappers won't have to worry about placing points perfectly and can oftentimes skip moving them altogether.

I also realized that the "Extract this point" validation fix for point-as-vertex warnings already works likes this, so there's precedent within iD. It's good making them consistent.
